### PR TITLE
feat: Implement GET `/medal/c/c:country_code` endpoint with `sub_sports`

### PR DESCRIPTION
**Pull request by Supakrit65**
- Add working implementation for GET /medal/c/c:country_code endpoint
- I actually tested it on the mock data, so it should be fine. **Trust me bro!**
Screenshot 2566-10-28 at 13 13 06

### OUTPUT
```
{
	"country_name": "United States of America",
	"gold": 40,
	"silver": 20,
	"bronze": 10,
	"individual_sports": [
		{
			"sport_id": 1,
			"sport_name": "Archery",
			"gold": 30,
			"silver": 20,
			"bronze": 10,
			"sub_sports": [
				{
					"sub_id": 1,
					"sub_name": "Individual Men's",
					"gold": 10,
					"silver": 10,
					"bronze": 0
				},
				{
					"sub_id": 2,
					"sub_name": "Individual Women's",
					"gold": 20,
					"silver": 10,
					"bronze": 10
				}
			]
		},
		{
			"sport_id": 2,
			"sport_name": "Artistic Gymnastics",
			"gold": 10,
			"silver": 0,
			"bronze": 0,
			"sub_sports": [
				{
					"sub_id": 3,
					"sub_name": "Individual All-Around Men's",
					"gold": 10,
					"silver": 0,
					"bronze": 0
				}
			]
		}
	],
	"country": "US"
}
```

### CODE REVIEW

**Review by RyukungG — DISMISSED** [review]
Nice!

**Review by MarojT — DISMISSED** [review]
Look great.

**Comment by Supakrit65** [comment]
Alt Text

File: sota/routers/medal_router.py [hunk 1]
```diff
@@ -31,9 +27,16 @@ def get_medal_by_country(country_code: str):
...
                 "silver": {"$sum": "$sports.silver"},
                 "bronze": {"$sum": "$sports.bronze"},
+                "sub_sports": {
+                    "$push": {
+                        "sub_id": "$sports.type_id",
+                        "gold": "$sports.gold",
+                        "silver": "$sports.silver",
+                        "bronze": "$sports.bronze",
+                    }
+                },
```
Comment (GToidZ) [inline sota/routers/medal_router.py]: I don't want to be a spoilsport, but I think you missed name of subsports.

**Review by GToidZ — CHANGES_REQUESTED** [review]

File: sota/routers/medal_router.py (hunk 1 above)
Comment (Supakrit65) [reply sota/routers/medal_router.py]: yeah totally

Comment (GToidZ) [reply sota/routers/medal_router.py]: Output

Affirmative. I checked with the API tester, and it doesn't contain `sub_sport_name`, please fix. 🙏

**Review by GToidZ — APPROVED** [review]
Good! I think you should merge #10 first, then this one. Otherwise, nice work!

**Review by RyukungG — APPROVED** [review]
Goodๆ

**Review by Halcyon905 — APPROVED** [review]
looks great

**Pull request merged**